input.go

```diff
@@ -7,10 +7,200 @@ import (
+		return cadenceVal, nil
+	}
+	f := reflect.ValueOf(v)
+	return ReflectToCadenceWithTypeHint(f, typeHint, resolver)
+}
+
+func ReflectToCadenceWithTypeHint(value reflect.Value, typeHint sema.Type, resolver InputResolver) (cadence.Value, error) {
+	if value == reflect.ValueOf(nil) {
+		return cadence.NewOptional(nil), nil
+	}
+	inputType := value.Type()
+
+	kind := inputType.Kind()
+	switch kind {
+	case reflect.Struct:
+		var val []cadence.Value
+		fields := []cadence.Field{}
+		for i := 0; i < value.NumField(); i++ {
+			fieldValue := value.Field(i)
+			// dont know how to have a sema.TypeHint here
+			cadenceVal, err := ReflectToCadence(fieldValue, resolver)
+			if err != nil {
+				return nil, err
+			}
+			cadenceType := cadenceVal.Type()
+
+			field := inputType.Field(i)
+
+			tags, err := structtag.Parse(string(field.Tag))
+			if err != nil {
+				return nil, err
+			}
+
+			name := ""
+			tag, err := tags.Get("cadence")
+			if err != nil {
+				tag, _ = tags.Get("json")
+			}
+			if tag != nil {
+				name = tag.Name
+			}
+
+			if name == "-" {
+				continue
+			}
+
+			if name == "" {
+				name = strings.ToLower(field.Name)
+			}
+
+			if IsTagCadecenAddress(tag) {
+				stringVal := getAndUnquoteString(cadenceVal)
+				adr, err := hexToAddress(stringVal)
+				if err != nil {
+					return nil, err
+				}
+				cadenceAddress := cadence.BytesToAddress(adr.Bytes())
+				cadenceType = cadence.AddressType
+				cadenceVal = cadenceAddress
+			}
+
+			fields = append(fields, cadence.Field{
+				Identifier: name,
+				Type:       cadenceType,
+			})
+
+			val = append(val, cadenceVal)
+		}
+
+		resolvedIdentifier, err := resolver(inputType.Name(), Identifier)
+		if err != nil {
+			return nil, err
+		}
+		structType := cadence.StructType{
+			QualifiedIdentifier: resolvedIdentifier,
+			Fields:              fields,
+		}
+
+		structValue := cadence.NewStruct(val).WithType(&structType)
+		return structValue, nil
+
+	case reflect.Pointer:
+		/*
+			*  //not sure when this will fire?
+						if value.IsNil() {
+							return cadence.NewOptional(nil), nil
+						}
+		*/
+
+		hint := typeHint.(*sema.OptionalType)
+		ptrValue, err := ReflectToCadenceWithTypeHint(value.Elem(), hint.Type, resolver)
+		if err != nil {
+			return nil, errors.Wrap(err, "getting value of optional type")
+		}
+		return cadence.NewOptional(ptrValue), nil
+
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64, reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return runtime.ParseLiteral(fmt.Sprintf("%v", value), typeHint, flowInterpeter)
+	case reflect.Bool:
+		return cadence.NewBool(value.Interface().(bool)), nil
+	case reflect.String:
+		stringVal := value.Interface().(string)
+		th := typeHint
+		optionalHint, isOptionalType := typeHint.(*sema.OptionalType)
+		if isOptionalType {
+			th = optionalHint.Type
+		}
+		if th == sema.TheAddressType {
+			// TODO: do we need to have different types of resolvers here?
```

Comment:
might be good to add this to our backlog story

Reply:
i have removed this as we do have types for resolvers now.